internal/resources/policy/kind/network/input_schema.go

```diff
@@ -0,0 +1,222 @@
+		if v1, ok := v.([]interface{}); ok && len(v1) != 0 {
+			inputRecipeData = &inputRecipe{
+				recipe:             CustomIngressRecipe,
+				inputCustomIngress: reciperesource.ConstructCustomIngress(v1),
+			}
+		}
+	}
+
+	return inputRecipeData
+}
+
+func flattenInput(inputRecipeData *inputRecipe) (data []interface{}) {
+	if inputRecipeData == nil {
+		return data
+	}
+
+	flattenInputData := make(map[string]interface{})
+
+	switch inputRecipeData.recipe {
+	case AllowAllRecipe:
+		flattenInputData[reciperesource.AllowAllKey] = reciperesource.FlattenAllowAll(inputRecipeData.inputAllowAll)
+	case AllowAllToPodsRecipe:
+		flattenInputData[reciperesource.AllowAllToPodsKey] = reciperesource.FlattenAllowAllToPods(inputRecipeData.inputAllowAllToPods)
+	case AllowAllEgressRecipe:
+		flattenInputData[reciperesource.AllowAllEgressKey] = []interface{}{make(map[string]interface{})}
+	case DenyAllRecipe:
+		flattenInputData[reciperesource.DenyAllKey] = []interface{}{make(map[string]interface{})}
+	case DenyAllToPodsRecipe:
+		flattenInputData[reciperesource.DenyAllToPodsKey] = reciperesource.FlattenDenyAllToPods(inputRecipeData.inputDenyAllToPods)
+	case DenyAllEgressRecipe:
+		flattenInputData[reciperesource.DenyAllEgressKey] = []interface{}{make(map[string]interface{})}
+	case CustomEgressRecipe:
+		flattenInputData[reciperesource.CustomEgressKey] = reciperesource.FlattenCustomEgress(inputRecipeData.inputCustomEgress)
+	case CustomIngressRecipe:
+		flattenInputData[reciperesource.CustomIngressKey] = reciperesource.FlattenCustomIngress(inputRecipeData.inputCustomIngress)
+	case UnknownRecipe:
+		fmt.Printf("[ERROR]: No valid input recipe block found: minimum one valid input recipe block is required among: %v. Please check the schema.", strings.Join(RecipesAllowed[:], `, `))
+	}
+
+	return []interface{}{flattenInputData}
+}
+
+func ValidateInput(_ context.Context, diff *schema.ResourceDiff, _ interface{}) error {
+	value, ok := diff.GetOk(policy.SpecKey)
+	if !ok {
+		return fmt.Errorf("spec: %v is not valid: minimum one valid spec block is required", value)
+	}
+
+	data, _ := value.([]interface{})
+
+	if len(data) == 0 || data[0] == nil {
+		return fmt.Errorf("spec data: %v is not valid: minimum one valid spec block is required among: %v", data, strings.Join(RecipesAllowed[:], `, `))
+	}
+
+	specData := data[0].(map[string]interface{})
+
+	input, ok := specData[policy.InputKey]
+	if !ok {
+		return fmt.Errorf("input: %v is not valid: minimum one valid input block is required", input)
+	}
+
+	inputType, ok := input.([]interface{})
+	if !ok {
+		return fmt.Errorf("type of input block data: %v is not valid", inputType)
+	}
+
+	if len(inputType) == 0 || inputType[0] == nil {
+		return fmt.Errorf("input data: %v is not valid: minimum one valid input block is required", inputType)
+	}
+
+	inputData, _ := inputType[0].(map[string]interface{})
+	recipesFound := make([]string, 0)
+
+	appendRecipeFromInput(inputData, &recipesFound, reciperesource.AllowAllKey)
```

Comment:
why cant we move all these appendRecipeFromInput() statements inside the appendRecipeFromInput function and append all the recipes at once ?

Reply:
Done